src/components/inventory.rs

```diff
@@ -140,24 +140,31 @@ impl GameEntity<(&mut Context, &mut Player, &mut Inventory, &mut HUD)> for Inven
             InventoryFocus::None => {
                 self.focus = InventoryFocus::Weapons;
                 state.control_flags.set_ok_button_disabled(false);
-                state.textscript_vm.start_script(self.get_weapon_event_number(inventory));
+                // check weapon count (0 count means we run item script)
+                state.textscript_vm.start_script( if self.weapon_count > 0 {self.get_weapon_event_number(inventory)} else {self.get_item_event_number(inventory)});
             }
             InventoryFocus::Weapons if state.control_flags.control_enabled() => {
-                if player.controller.trigger_left() {
-                    state.sound_manager.play_sfx(4);
-                    inventory.prev_weapon();
-                    state.control_flags.set_ok_button_disabled(false);
-                    state.textscript_vm.start_script(self.get_weapon_event_number(inventory));
-                }
 
-                if player.controller.trigger_right() {
-                    state.sound_manager.play_sfx(4);
-                    inventory.next_weapon();
-                    state.control_flags.set_ok_button_disabled(false);
-                    state.textscript_vm.start_script(self.get_weapon_event_number(inventory));
+                // if we have no weapons, the TSC should not be refreshed with L/R keystrokes
+                if self.weapon_count > 0
+                {
+                    if player.controller.trigger_left() {
```

Comment:
That would become problematic. 
The result would be these lines of code running on every single tick that the inventory is open, instead of just when a key is pressed.
```
state.sound_manager.play_sfx(4);
state.control_flags.set_ok_button_disabled(false);
state.textscript_vm.start_script(self.get_weapon_event_number(inventory));
```
The only difference in this section between my alterations and the original is nesting the left and right press events inside a check for weapons in the inventory (to stop them from running at all if the player doesn't have any guns).

Without some third general key check, moving the non-unique events outside the left and right functions will result in them being run every tick. *(and adding this key check adds a layer of unnecessary complexity, especially since we still have to check for individual left and right keys anyway.)*